components/encryption/src/crypter.rs

```diff
@@ -56,41 +56,27 @@ pub fn get_method_key_length(method: EncryptionMethod) -> usize {
     }
 }
 
-pub enum Cipher {
-    Plaintext,
-    AesCtr(OCipher),
-}
-
-impl From<EncryptionMethod> for Cipher {
-    fn from(method: EncryptionMethod) -> Cipher {
-        match method {
-            EncryptionMethod::Plaintext => Cipher::Plaintext,
-            EncryptionMethod::Aes128Ctr => Cipher::AesCtr(OCipher::aes_128_ctr()),
-            EncryptionMethod::Aes192Ctr => Cipher::AesCtr(OCipher::aes_192_ctr()),
-            EncryptionMethod::Aes256Ctr => Cipher::AesCtr(OCipher::aes_256_ctr()),
-            unknown => panic!("bad EncryptionMethod {:?}", unknown),
-        }
-    }
-}
-
-// IV as an AES input, the length must be 16 btyes.
-const IV_LEN: usize = 16;
+// IV as an AES input, the length should be 12 btyes for GCM mode.
+const GCM_IV_LEN: usize = 12;
 
 #[derive(Debug, Clone, Copy)]
 pub struct Iv {
-    iv: [u8; IV_LEN],
+    iv: [u8; GCM_IV_LEN],
```

Comment:
let's make a new `IV12` type which is of 12 bytes. we still need to use 16 bytes IV for snapshot and backup encryption.